Use fmt::format for debug strings in Library init

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151682
* #151697
* #151630
* __->__ #151629
* #151628
* #151627
* #151626

Observed several ms taken during `import torch` by c10::str here.

Differential Revision: [D73129453](https://our.internmc.facebook.com/intern/diff/D73129453/)

swolchok committed:
Use fmt::format for debug strings in Library init

Observed several ms taken during `import torch` by c10::str here.

Differential Revision: [D73129453](https://our.internmc.facebook.com/intern/diff/D73129453/)

[ghstack-poisoned]

cyyever commented:
Good job. Can we apply libfmt in more ``.pp`` files

Skylion007 commented:
> Good job. Can we apply libfmt in more `.cpp` files?

@cyyever I would def by open to more PRs that do this.

swolchok committed:
Update on "Use fmt::format for debug strings in Library init"


Observed several ms taken during `import torch` by c10::str here.

Differential Revision: [D73129453](https://our.internmc.facebook.com/intern/diff/D73129453/)

[ghstack-poisoned]
